### ISSUE TITLE
`no-abusive-eslint-disable` can’t be disabled

### ISSUE THREAD

**Issue by patrik-csak** [sindresorhus/eslint-plugin-unicorn#2624]
Context: I encounter this issue when using `eslint-plugin-unicorn@58.0.0` via `xo@0.61.0-1`

---

What’s wrong: `unicorn/no-abusive-eslint-disable` can’t be disabled with an ESLint configuration comment, because the `eslint-disable unicorn/no-abusive-eslint-disable` comment is incorrectly identified as unused.

Say I have some messy code:

```js
function messyCode() {
    // ...
}
```

I’m not ready to address it, so I disable ESLint for that block of code:

```js
/* eslint-disable */
function messyCode() {
    // ...
}
/* eslint-enable */
```

This triggers the `no-abusive-eslint-disable` error (‘Specify the rules you want to disable’). So I add an ESLint comment to disable the `no-abusive-eslint-disable` rule:

```js
/* eslint-disable unicorn/no-abusive-eslint-disable */

/* eslint-disable */
function messyCode() {
    // ...
}
/* eslint-enable */
```

This triggers an ESLint *warning*:

> ESLint: Unused eslint-disable directive (no problems were reported from 'unicorn/no-abusive-eslint-disable’).

The directive is not unused. If I run `eslint`, I no longer get the `no-abusive-eslint-disable` error. But if I run `eslint --fix`, the unused directive will be deleted, and the `no-abusive-eslint-disable` error will return.

---

I didn’t have the same issue disabling [`eslint-comments/no-unlimited-disable`](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/v3.2.0/docs/rules/no-unlimited-disable.md). This worked:

```js
/* eslint-disable eslint-comments/no-unlimited-disable */

/* eslint-disable */
function messyCode() {
    // ...
}
/* eslint-enable */
```

**Comment by watson** [comment]
I encountered the same issue in `eslint-plugin-unicorn@59.0.1`. I can confirm that the issue doesn't exist in `eslint-plugin-unicorn@57.0.0` (which is the version we're trying to upgrade from)
